### ISSUE TITLE
Incorrect Display Ratio ? / Ratio Option Suggestion

### ISSUE THREAD

**Issue by peeweek** [spiritualized1997/openFPGA-NES#14]
I have noticed that all my NES games tend to stretch to a 4:3 aspect ratio with black bars on top/bottom of the screen. I was not able to measure exactly the ratio on screen but it seems that it does not follow the original aspect of the console.

![image](https://user-images.githubusercontent.com/4037271/209934045-1ba03055-0741-4b78-9110-cbc89f96210c.png)

If I am not mistaken, NES (PAL) Displays as 256×240 (16:15 ratio) and NES (NTSC) displays as 256×224 (8:7 ratio)

Given the ratio of the screen 1600×1440 (10:9), correct ratio resolution values could be :

Integer scaling (6x) PAL
- 1536 pixels wide (padded at integer 6x)
- 1440 pixels tall (for PAL 240) 
- Slight Black bars left and right

Integer scaling (6x) NSTC
- 1536 pixels wide (padded at integer 6x)
- 1344 pixels tall (for NTSC 224)
- Slight Black padding around image

Non-integer-scaling (6.25x NTSC)
- 1600 pixels wide/ 1400 pixels tall for NTSC (20 px black bars top+bottom)

I know that screen ratio is a divisive debate, maybe adding an option to select screen ratios could be a great addition to the core.

**Comment by ematima** [comment]
I have seen the same behavior happening.  Additionally, I am seeing pixels being cut off of the bottom of the game I am playing, Elite.  The behavior is different though depending on how I am looking at the game:

When playing on the pocket, the image is cropped along the bottom and the aspect ratio looks wrong:
![EliteOnPocket](https://github.com/spiritualized1997/openFPGA-NES/assets/5911415/86577711-1f47-4e56-a548-0e9d46d52f33)

When I do a screen shot of the game on the pocket, the screenshot looks correct:
![20231018_135004](https://github.com/spiritualized1997/openFPGA-NES/assets/5911415/f7f64069-608d-40d4-a034-b33c07a8d8bb)

When docked, the image is not cropped, but aspect still looks off?:
![eliteDocked](https://github.com/spiritualized1997/openFPGA-NES/assets/5911415/33610f3c-42b5-4b9b-bbd8-afd34da3cef2)

**Comment by SegaSnatcher** [comment]
> I have noticed that all my NES games tend to stretch to a 4:3 aspect ratio with black bars on top/bottom of the screen. I was not able to measure exactly the ratio on screen but it seems that it does not follow the original aspect of the console.
> 
> 
> 
> If I am not mistaken, NES (PAL) Displays as 256×240 (16:15 ratio) and NES (NTSC) displays as 256×224 (8:7 ratio)
> 
> Given the ratio of the screen 1600×1440 (10:9), correct ratio resolution values could be :
> 
> Integer scaling (6x) PAL
> 
> * 1536 pixels wide (padded at integer 6x)
> * 1440 pixels tall (for PAL 240)
> * Slight Black bars left and right
> 
> Integer scaling (6x) NSTC
> 
> * 1536 pixels wide (padded at integer 6x)
> * 1344 pixels tall (for NTSC 224)
> * Slight Black padding around image
> 
> Non-integer-scaling (6.25x NTSC)
> 
> * 1600 pixels wide/ 1400 pixels tall for NTSC (20 px black bars top+bottom)
> 
> I know that screen ratio is a divisive debate, maybe adding an option to select screen ratios could be a great addition to the core.

The Pocket's LCD aspect ratio is 10:9 which is what the original Gameboy LCD aspect ratio is.  NES display aspect ratio is not 10:9, but really closer to 64:49 which is just tad bit wider than 4:3. 

While NES has a pixel resolution of 8:7, the actual display aspect ratio on a CRT is 64:49.  

TLDR: Black bars on top and bottom are normal and what is accurate to real hardware.  Black Bars on the left and right on a 10:9 screen would not be accurate to NES's actual display aspect ratio. 

Receipts. 

![Receipts2](https://github.com/spiritualized1997/openFPGA-NES/assets/36483667/545e227c-a146-49df-b825-d310b77ff3c7)

**Comment by eltees** [comment]
yeah as SegaSnatcher mentioned, the final output (display aspect ratio, 8:7) is a combination of the device ratio (pixels wide by high) and the pixel aspect ratio (how 'square' or un square each pixel is.  In the world of SD television broadcast standards, pixels are not actually square but oblong i.e. wider than they are tall, this is where the *second* 8:7 ratio comes in, multiplying the two together gives you 8^2/7^2 hence the 64:49 final display ratio which is extremely close to 4:3 (which would be 64:48)
